terraso_backend/apps/graphql/schema/commons.py

```diff
@@ -43,6 +44,35 @@ def resolve_total_count(self, info, **kwargs):
         queryset = self.iterable
         return queryset.count()
 
+    # This will coax graphene to output more precise types for connections.
+    # Context: https://github.com/graphql-python/graphene/pull/1504
+    # Will be unnecessary after https://github.com/graphql-python/graphene-django/issues/901
+    @classmethod
+    def __init_subclass_with_meta__(cls, node=None, **options):
+        type_name = re.sub("Connection$", "", cls.__name__)
+
+        node_for_edge = node
+        if node is not None and not isinstance(node, NonNull):
+            node_for_edge = NonNull(node)
+
+        class Edge(ObjectType):
+            node = Field(node_for_edge, description="The item at the end of the edge")
+            cursor = String(required=True, description="A cursor for use in pagination")
+
+        class Meta:
+            description = f"A Relay edge containing a `{type_name}` and its cursor."
+
+        edge_type = type(f"{type_name}Edge", (Edge,), {"Meta": Meta})
+
+        cls.Edge = edge_type
+
+        cls.edges = Field(
+            NonNull(List(NonNull(edge_type))),
+            description="Contains the nodes in this connection.",
+        )
+
+        super(TerrasoConnection, cls).__init_subclass_with_meta__(node=node, **options)
```

Comment:
suggestion: I *think* it is possible to just use [super](https://docs.python.org/3/library/functions.html#super) here, which I find a bit less confusing.